### ISSUE TITLE
$NINJA_PATH/ninja step error issue

### ISSUE THREAD

**Issue by kkhsuk** [thomaspark-pkj/darknet-nnpack#60]
when I install the darknet-nnpack

I have an issue on step of "$NINJA_PATH"

[62/147] LINK bin/transform-benchmark
FAILED: /home/pi/ninja/NNPACK-darknet/bin/transform-benchmark 
g++ -pthread  -o /home/pi/ninja/NNPACK-darknet/bin/transform-benchmark /home/pi/ninja/NNPACK-darknet/build/bench/transform.c.o /home/pi/ninja/NNPACK-darknet/build/bench/median.c.o /home/pi/ninja/NNPACK-darknet/build/bench/memread.c.o -lrt /home/pi/ninja/NNPACK-darknet/lib/libnnpack.a /home/pi/ninja/NNPACK-darknet/lib/libpthreadpool.a 
/home/pi/ninja/NNPACK-darknet/lib/libpthreadpool.a(pthreads.c.o): In function `pthreadpool_create':
/home/pi/ninja/NNPACK-darknet/deps/pthreadpool/src/pthreads.c:258: undefined reference to `pthreadpool_allocate'
/home/pi/ninja/NNPACK-darknet/lib/libpthreadpool.a(pthreads.c.o): In function `pthreadpool_destroy':
/home/pi/ninja/NNPACK-darknet/deps/pthreadpool/src/pthreads.c:459: undefined reference to `pthreadpool_deallocate'
/usr/bin/ld: /home/pi/ninja/NNPACK-darknet/bin/transform-benchmark: internal symbol `pthreadpool_deallocate' isn't defined
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
[63/147] LINK bin/convolution-benchmark
FAILED: /home/pi/ninja/NNPACK-darknet/bin/convolution-benchmark 
g++ -pthread  -o /home/pi/ninja/NNPACK-darknet/bin/convolution-benchmark /home/pi/ninja/NNPACK-darknet/build/bench/convolution.c.o /home/pi/ninja/NNPACK-darknet/build/bench/median.c.o /home/pi/ninja/NNPACK-darknet/build/bench/memread.c.o -lrt /home/pi/ninja/NNPACK-darknet/lib/libnnpack.a /home/pi/ninja/NNPACK-darknet/lib/libpthreadpool.a 
/home/pi/ninja/NNPACK-darknet/lib/libpthreadpool.a(pthreads.c.o): In function `pthreadpool_create':
/home/pi/ninja/NNPACK-darknet/deps/pthreadpool/src/pthreads.c:258: undefined reference to `pthreadpool_allocate'
/home/pi/ninja/NNPACK-darknet/lib/libpthreadpool.a(pthreads.c.o): In function `pthreadpool_destroy':
/home/pi/ninja/NNPACK-darknet/deps/pthreadpool/src/pthreads.c:459: undefined reference to `pthreadpool_deallocate'
/usr/bin/ld: /home/pi/ninja/NNPACK-darknet/bin/convolution-benchmark: internal symbol `pthreadpool_deallocate' isn't defined
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
[64/147] CXX test/convolution-output/alexnet+relu.cc
ninja: build stopped: subcommand failed.

somebody help me

**Comment by Shivankit99** [comment]
Just avoid using Ninja and build NNPACK using cmake and make install.

**Comment by GiorgioSgl** [comment]
Same here, as you solved in some way?

**Comment by kkhsuk** [comment]
someone send to me a this solution 
 
Just avoid using Ninja and build NNPACK using cmake and make install. 
 
this solution is working but we had some problem of the darknet-nnpack so we didn't use darknet-nnpack

-----Original Message-----

**Comment by GiorgioSgl** [comment]
> someone send to me a this solution Just avoid using Ninja and build NNPACK using cmake and make install. this solution is working but we had some problem of the darknet-nnpack so we didn't use darknet-nnpack
> […](#)
> -----Original Message----- From: "Giorgio Segalla"<notifications@github.com> To: "digitalbrain79/darknet-nnpack"<darknet-nnpack@noreply.github.com>; Cc: "kkhsuk"<kkhsuk@naver.com>; "Author"<author@noreply.github.com>; Sent: 2021-03-10 (수) 00:21:46 (GMT+09:00) Subject: Re: [digitalbrain79/darknet-nnpack] $NINJA_PATH/ninja step error issue (#60) Same here, as you solved in some way? — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Okay seem I solved it follow your suggestion, thank you very much. 

These are the command for who doesn't know hot to run cmake with make, like me ahhahah

All these in NNPACK directory
`mkdir build`
`cd build`
`cmake -G "Unix Makefiles" ..`
`make`

**Comment by GiorgioSgl** [comment]
After I get an error when try to build the darknet-nnpack
I create a new [issue](https://github.com/digitalbrain79/darknet-nnpack/issues/63#issue-827579417) try to check it got solved before test the comment above

**Comment by kkhsuk** [comment]
I didn't see this issue by my rasp pi 3B
 
after the solved cmake issue and we gave up use darknet-nnpack
when I send the email to digitalbrain79 he said try to use the rasp pi os image in the Readme file link
and he said darknet-nnpack is old system so he suggest the another framework like tensorflow
 
when i gave up the darknet-nnpack, I use the tf lite withe the google coral usb accelerator
 
this is the only advice I can give you
 
-----Original Message-----

**Comment by GiorgioSgl** [comment]
> I didn't see this issue by my rasp pi 3B after the solved cmake issue and we gave up use darknet-nnpack when I send the email to digitalbrain79 he said try to use the rasp pi os image in the Readme file link and he said darknet-nnpack is old system so he suggest the another framework like tensorflow when i gave up the darknet-nnpack, I use the tf lite withe the google coral usb accelerator this is the only advice I can give you
> […](#)
> -----Original Message----- From: "Giorgio Segalla"<notifications@github.com> To: "digitalbrain79/darknet-nnpack"<darknet-nnpack@noreply.github.com>; Cc: "kkhsuk"<kkhsuk@naver.com>; "Author"<author@noreply.github.com>; Sent: 2021-03-10 (수) 20:29:38 (GMT+09:00) Subject: Re: [digitalbrain79/darknet-nnpack] $NINJA_PATH/ninja step error issue (#60) After I get an error when try to build the darknet-nnpack I create a new issue try to check it got solved before test the comment above — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Yeah I agree, but I have already implemented yolov4-tiny in the tflite and I got very bad result in the mAP50 (29%) while by state of the art it would get 40%. That's the main reason I was trying to use the darknetNNPACK, because with the normal version of darknet I got 0.1 FPS very bad result. I will try somenew things and hopefully get some better result.
